pkg/api/kubernetesagentpool/validate.go

```diff
@@ -0,0 +1,99 @@
+package kubernetesagentpool
+
+import (
+	"fmt"
+	"strings"
+)
+
+func (p *AgentPool) Validate() error {
```

Comment:
It could be for sure, is that out of scope though? I feel like we could have an error-message beautifying PR at another time?